c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -585,14 +700,26 @@ class DeviceCachingAllocator {
         garbage_collect_cached_blocks();
       }
       // Attempt allocate
-      block_found = alloc_block(params, false)
-          // Free enough available cached blocks to satisfy alloc and retry
-          // alloc.
-          || (release_available_cached_blocks(params) &&
-              alloc_block(params, false))
-          // Free all non-split cached blocks and retry alloc.
-          || (C10_LIKELY(captures_underway == 0) && release_cached_blocks() &&
-              alloc_block(params, true));
+      block_found = alloc_block(params, false);
```

Comment:
Do you plan to allow users to toggle the new algorithm instead of forcing it for all allocations?

Reply:
Done. Thank you.